mm2src/lp_ordermatch.rs

```diff
@@ -114,26 +115,76 @@ async fn process_order_keep_alive(
     let ordermatch_ctx = OrdermatchContext::from_ctx(&ctx).unwrap();
     let mut orderbook = ordermatch_ctx.orderbook.lock().await;
     let uuid = keep_alive.uuid.into();
-    match find_order_by_uuid_and_pubkey(&mut orderbook, &uuid, from_pubkey) {
-        Some(order) => {
-            order.timestamp = keep_alive.timestamp;
-            true
-        },
-        None => match ordermatch_ctx.inactive_orders.lock().await.remove(&uuid) {
-            Some(mut order) => {
+    if let Some(order) = find_order_by_uuid_and_pubkey(&mut orderbook, &uuid, from_pubkey) {
+        order.timestamp = keep_alive.timestamp;
+        return true;
+    }
+
+    if let Some(mut order) = ordermatch_ctx.inactive_orders.lock().await.remove(&uuid) {
+        order.timestamp = keep_alive.timestamp;
+        orderbook
+            .entry((order.base.clone(), order.rel.clone()))
+            .or_insert_with(HashMap::new)
+            .insert(uuid, order);
+        return true;
+    }
+
+    log!("Couldn't find an order " [uuid] ", try request it from peers");
+    match request_order(ctx.clone(), uuid, from_pubkey).await {
+        Ok(Some(mut order)) => {
+            if order.timestamp < keep_alive.timestamp {
+                // a peer may not received the keep_alive yet
                 order.timestamp = keep_alive.timestamp;
-                orderbook
-                    .entry((order.base.clone(), order.rel.clone()))
-                    .or_insert_with(HashMap::new)
-                    .insert(uuid, order);
-                true
-            },
-            None => {
-                broadcast_repeat_order(ctx, topic.into(), keep_alive.uuid.into());
-                false
-            },
+            }
+            insert_or_update_order(&ctx, order, uuid).await;
+            return true;
         },
+        Ok(None) => log!("None of peers responded to the GetOrder request"),
+        Err(e) => log!("Error on GetOrder request: "(e)),
+    }
+
+    log!("try to broadcast the RepeatOrder");
+    broadcast_repeat_order(ctx, topic.into(), keep_alive.uuid.into());
+    false
+}
+
+async fn request_order(ctx: MmArc, uuid: Uuid, from_pubkey: &str) -> Result<Option<PricePingRequest>, String> {
+    let get_order = OrdermatchRequest::GetOrder {
+        uuid,
+        from_pubkey: from_pubkey.to_string(),
+    };
+    let req = P2PRequest::Ordermatch(get_order);
+    let response = match try_s!(request_any_peer(ctx, req).await) {
+        Some((response, _pubkey)) => response,
+        None => return Ok(None),
+    };
+
+    let (initial_message, from_peer) = match response {
```

Comment:
When we are about to send a request we know exactly what data type we should get in response. So P2PResponse enum can be removed in favor of making `request_any_peer` generic over response type. So we will be able to write
```
    let response:  OrderInitialMessage = match try_s!(request_any_peer(ctx, req).await) {
        Some((response, _pubkey)) => response,
        None => return Ok(None),
    };
```
In this case we won't need these `match` statements everywhere we need to perform a request.